integration-tests/chain-signatures/tests/actions/wait.rs

```diff
@@ -0,0 +1,255 @@
+use std::future::{Future, IntoFuture};
+
+use anyhow::Context;
+use backon::{ConstantBuilder, ExponentialBuilder, Retryable};
+use mpc_contract::{ProtocolContractState, RunningContractState};
+use mpc_node::web::StateView;
+
+use crate::cluster::Cluster;
+
+type Epoch = u64;
+
+enum WaitActions {
+    Running(Epoch),
+    MinTriples(usize),
+    MinMineTriples(usize),
+    MinPresignatures(usize),
+    MinMinePresignatures(usize),
+    Signable(usize),
```

Comment:
It is not clear what Signable means. How is it different from Running?

Reply:
How is it different from having 1  presignature?

Reply:
So right now, it requires X number of presignatures but we can have further conditions down the line for it. It's just from a usability standpoint, when you write `signable`, you don't care about what is going on underneath and only care that we can sign something